### PR TITLE
Ignore dg limited fields for residual

### PR DESCRIPTION
**Pull request by gassmoeller**
This PR depends on #5702 and #5531.

@anne-glerum this is a solution for the convergence issues you reported in #5531. We should make sure to merge the other two PRs first before this one to be able to test if it works.

### CODE REVIEW

File: source/simulator/solver.cc [hunk 1]
```diff
@@ -534,8 +534,15 @@ namespace aspect
         ||
         (!advection_field.is_temperature()
          && parameters.use_discontinuous_composition_discretization
-         && parameters.use_limiter_for_discontinuous_composition_solution))
-      apply_limiter_to_dg_solutions(advection_field);
+         && parameters.use_limiter_for_discontinuous_composition_solution[advection_field.compositional_variable]))
+      {
+        apply_limiter_to_dg_solutions(advection_field);
+        // by applying the limiter we have modified the solution to not longer
```
Comment (bangerth) [inline source/simulator/solver.cc:540]:
```suggestion
        // by applying the limiter we have modified the solution to no longer
```

**Comment by tjhei** [comment]
You will need to update the test output.

**Review by tjhei — APPROVED** [review]

**Comment by gassmoeller** [comment]
Great, this is exactly the change in test results we were hoping for. The test that @anne-glerum added in #5531 is nicely converging now. I updated the test results.

**Review by bangerth — APPROVED** [review]
This is not satisfying, but it's the right thing to do in view of the situation we find ourselves in. In a better world, the limiter would have been written in such a way that it appears in the weak formulation of the equation and consequently in the residual :-(

**Pull request merged**